src/frontend/pages/report-history.vue

```diff
@@ -11,25 +11,19 @@
     </v-row>
     <v-row v-else>
       <v-col v-for="report in reports" :id="report.id" :key="report.id" cols="12" class="pa-2">
-        <Success v-if="!report.data.error && report.data.done" :report="report" />
-        <In-progress v-if="!report.data.error && !report.data.done" :report="report" />
-        <Error v-if="report.data.error" :report="report" />
+        <Card :report="report" :status="getStatusOfReport(report)" />
```

Comment:
Methods are not meant to be used in render. You should use computed properties in the `Card` component instead. `Card` is probably too generic btw, something like `ReportCard` would be clearer.